packages/app-store/ee/routing-forms/components/react-awesome-query-builder/widgets.tsx

```diff
@@ -25,49 +79,70 @@ const TextAreaWidget = (props: TextWidgetProps) => {
 
   const textValue = value || "";
   return (
-    <TextArea
+    <textarea
       value={textValue}
       placeholder={placeholder}
-      disabled={readonly}
+      disabled={readOnly}
       onChange={onChange}
       maxLength={maxLength}
-      className="dark:border-darkgray-300 flex flex-grow border-gray-300 text-sm dark:bg-transparent dark:text-white dark:selection:bg-green-500 disabled:dark:text-gray-500"
+      className="dark:placeholder:text-darkgray-600 focus:border-brand dark:border-darkgray-300 dark:text-darkgray-900 block w-full rounded-md border-gray-300 text-sm focus:ring-black disabled:bg-gray-200 disabled:hover:cursor-not-allowed dark:bg-transparent dark:selection:bg-green-500 disabled:dark:text-gray-500"
       {...customProps}
       {...remainingProps}
     />
   );
 };
 
-const TextWidget = (props: TextWidgetProps & { type?: string }) => {
-  const { value, setValue, readonly, placeholder, customProps, ...remainingProps } = props;
-  let { type } = props;
-  type = type || "text";
+const TextWidget = (props: TextLikeComponentPropsRAQB) => {
+  const {
+    value,
+    noLabel,
+    setValue,
+    readOnly,
+    placeholder,
+    customProps,
+    type = "text",
+    ...remainingProps
+  } = props;
   const onChange = (e: ChangeEvent<HTMLInputElement>) => {
     const val = e.target.value;
     setValue(val);
   };
   const textValue = value || "";
+  if (noLabel) {
+    return (
+      <Input
+        type={type}
+        className="dark:placeholder:text-darkgray-600 focus:border-brand dark:border-darkgray-300 dark:text-darkgray-900 block w-full rounded-md border-gray-300 text-sm focus:ring-black disabled:bg-gray-200 disabled:hover:cursor-not-allowed dark:bg-transparent dark:selection:bg-green-500 disabled:dark:text-gray-500"
+        value={textValue}
+        placeholder={placeholder}
+        disabled={readOnly}
+        onChange={onChange}
+        {...remainingProps}
+        {...customProps}
+      />
+    );
+  }
```

Comment:
we could simply this by just adding 

`labelSrOnly={noLabel}` to the text field below. This will visually disable the label for the TextField but also still provide the input name to the htmlFor so we still support the a11y requirements here 

(If im not understanding whats happening here let me know :D )

Reply:
@sean-brydon This is exactly what I need. Completely missed it. Thanks 🙏